Lib/test/test_inspect.py

```diff
@@ -3674,6 +3674,41 @@ def test_signature_on_noncallable_mocks(self):
                 with self.assertRaises(TypeError):
                     inspect.signature(mock)
 
+    def test_signature_on_code_objects(self):
+        def func1(
+            a, b: int,
+            /,
+            c, d: str = 'a',
+            *args: int,
+            e, f: bool = True,
+            **kwargs: str,
+        ) -> float:
+            ...
+
+        def func2(a: str, b: int = 0, /): ...
+        def func3(): ...
+        def func4(*a, **k): ...
+        def func5(*, kw=False): ...
+
+        known_sigs = {
+            func1: '(a, b, /, c, d, *args, e, f, **kwargs)',
+            func2: '(a, b, /)',
+            func3: '()',
+            func4: '(*a, **k)',
+            func5: '(*, kw)',
+        }
+
+        for test_func, expected_sig in known_sigs.items():
+            with self.subTest(test_func=test_func, expected_sig=expected_sig):
+                self.assertEqual(
+                    str(inspect.Signature.from_code(test_func.__code__)),
+                    expected_sig,
+                )
+
+    def test_signature_on_non_code_objects(self):
+        with self.assertRaisesRegex(TypeError, '1 is not a code object'):
```

Comment:
Just move this test in the previous function, i don't think that it's worth it to separate it.